code/modules/mob/living/carbon/human/species.dm

```diff
@@ -1244,50 +1244,87 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 		SEND_SIGNAL(target, COMSIG_HUMAN_DISARM_HIT, user, user.zone_selected)
-		var/obj/item/bodypart/affecting = target.get_bodypart(randomized_zone)
-		var/randn = rand(1, 100)
-		if(randn <= 25)
-			playsound(target, 'sound/weapons/thudswoosh.ogg', 50, 1, -1)
-			target.visible_message("<span class='danger'>[user] has pushed [target]!</span>",
-				"<span class='userdanger'>[user] has pushed [target]!</span>", null, COMBAT_MESSAGE_RANGE)
-			target.apply_effect(40, EFFECT_PARALYZE, target.run_armor_check(affecting, "melee", "Your armor prevents your fall!", "Your armor softens your fall!"))
-			target.forcesay(GLOB.hit_appends)
-			log_combat(user, target, "pushed over")
-			return
-
-		if(randn <= 60)
-			var/obj/item/I = null
-			if(target.pulling)
-				target.visible_message("<span class='warning'>[user] has broken [target]'s grip on [target.pulling]!</span>")
-				target.stop_pulling()
-			else
-				I = target.get_active_held_item()
-				if(target.dropItemToGround(I))
-					target.visible_message("<span class='danger'>[user] has disarmed [target]!</span>", \
-						"<span class='userdanger'>[user] has disarmed [target]!</span>", null, COMBAT_MESSAGE_RANGE)
+		//var/obj/item/bodypart/affecting = target.get_bodypart(randomized_zone)
+		var/shove_dir = get_dir(user.loc, target.loc)
+		var/turf/target_shove_turf = get_step(target.loc, shove_dir)
+		var/shove_blocked = FALSE
+		if (shove_dir in GLOB.diagonals)
+			var/turf/target_loc = target.loc
+			if(!target_loc.ClickCross(shove_dir, FALSE, target, target))
+				var/turf/cardinal_turf_1 = get_step(user.loc, turn(shove_dir, -45))
+				var/turf/cardinal_turf_2 = get_step(user.loc, turn(shove_dir, 45))
+				var/free_tile_found = FALSE
+				if(prob(50)) //Check the two turfs in a random order to make sure if both are free one isn't always favored
+					if(!is_blocked_turf(cardinal_turf_2, FALSE))
+						target_shove_turf = cardinal_turf_2
+						free_tile_found = TRUE
+					if(free_tile_found && !is_blocked_turf(cardinal_turf_1, FALSE))
+						target_shove_turf = cardinal_turf_1
+						free_tile_found = TRUE
+					if(!free_tile_found)
+						shove_blocked = TRUE
 				else
-					I = null
-			playsound(target, 'sound/weapons/thudswoosh.ogg', 50, 1, -1)
-			log_combat(user, target, "disarmed", "[I ? " removing \the [I]" : ""]")
-			return
-
-
-		playsound(target, 'sound/weapons/punchmiss.ogg', 25, 1, -1)
-		target.visible_message("<span class='danger'>[user] attempted to disarm [target]!</span>", \
-						"<span class='userdanger'>[user] attempted to disarm [target]!</span>", null, COMBAT_MESSAGE_RANGE)
-		log_combat(user, target, "attempted to disarm")
-
+					if(is_blocked_turf(cardinal_turf_1, FALSE))
+						target_shove_turf = cardinal_turf_1
+						free_tile_found = TRUE
+					if(free_tile_found && is_blocked_turf(cardinal_turf_2, FALSE))
+						target_shove_turf = cardinal_turf_2
+						free_tile_found = TRUE
+					if(!free_tile_found)
+						shove_blocked = TRUE
```

Comment:
Personally I would have the prob() be in assigning cardinal_turf_1 and cardinal_turf_2 so you don't have to duplicate the logic. Something like:
var/turf/cardinal_turf_1
var/turf/cardinal_turf_2
if(prob(50))
    cardinal_turf_1 = get_step(user.loc, turn(shove_dir, 45))
    cardinal_turf_2 = get_step(user.loc, turn(shove_dir, -45))
else
    cardinal_turf_1 = get_step(user.loc, turn(shove_dir, -45))
    cardinal_turf_2 = get_step(user.loc, turn(shove_dir, 45));

Then you just check both cards with no care which order they're in.

Reply:
Good idea